### DISCUSSION TITLE
Features and Improvements

### DISCUSSION THREAD

**Discussion by nreinicke** [NREL/mappymatch#2] [Ideas]
yamm is in the relatively early stages of development and so there are several potential features / improvements. This discussion serves as a central collection of these.

## Documentation
 - [ ] create developer documentation
 - [ ] create general documentation (setup, quick start, etc)
 - [ ] update doc strings on classes / functions
 - [ ] generate API documentation

## Installation
 - [ ] allow the package to be installed solely via pip 
 - [ ] distribute the package via PyPi and/or conda-forge

## Core 
 - [ ] load a trace from a gpx file
 - [ ] support other OSM networks besides drive

## Code Style
 - [ ] make sure typing hints are prevalent
 - [ ] resolve any mypy errors

## Testing
- [ ] add unit tests for the `LCSSMatcher` 
- [ ] add functional tests that define accuracy of map matching

## Interfaces
 - [ ] add interface to the Valhalla map matching API
 
## Algorithms
 - [ ] implement a Hidden Markov Model matching algorithm

## Visualization
 - [ ] improve visualization of matches (show line connecting point to matching road)

**Comment by Rowlando13** [comment]
Documentation: 
I really like restructured text. It has really nice auto documentation functions. You can also set webhooks to rebuild on commit or other action which I really like. It also supports versioned documentation which I like. [This](https://the-renamer.readthedocs.io/en/latest/) is a toy project I did for reference. 
For form factor, I like to keep the readme short and to the point so it does not require a lot of updating. I point directly at the docs. [Here](https://github.com/Rowlando13/the_renamer) is a sample of what I had in mind.

**Comment by Rowlando13** [comment]
Code Style 
Suggested integrations as Precommit actions: 

- black - just not worry about style
- isort - not worry about import styling or order 
- flake8 - helps catch hidden errors 

Testing 
- Implement an automated action to run the tests on a pull request for branch merge

**Comment by Rowlando13** [comment]
As far as developer docs, I really like [pyjanitor's documentation](https://pyjanitor-devs.github.io/pyjanitor/devguide/). It is very straightforward for newcomers.
